osu.Game.Rulesets.Osu/Difficulty/OsuDifficultyCalculator.cs

```diff
@@ -81,11 +81,26 @@ protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beat
 
             double preempt = IBeatmapDifficultyInfo.DifficultyRange(beatmap.Difficulty.ApproachRate, 1800, 1200, 450) / clockRate;
             double drainRate = beatmap.Difficulty.DrainRate;
-            int maxCombo = beatmap.GetMaxCombo();
 
-            int hitCirclesCount = beatmap.HitObjects.Count(h => h is HitCircle);
-            int sliderCount = beatmap.HitObjects.Count(h => h is Slider);
-            int spinnerCount = beatmap.HitObjects.Count(h => h is Spinner);
+            int maxCombo;
+            int hitCirclesCount, sliderCount, spinnerCount;
+
+            if (beatmap is ProgressiveCalculationBeatmap pcBeatmap)
+            {
+                maxCombo = pcBeatmap.GetMaxCombo();
+
+                hitCirclesCount = pcBeatmap.GetHitObjectCountOf(typeof(HitCircle));
+                sliderCount = pcBeatmap.GetHitObjectCountOf(typeof(Slider));
+                spinnerCount = pcBeatmap.GetHitObjectCountOf(typeof(Spinner));
+            }
```

Comment:
I will 2000000000% not allow this to pass review. Find another way around it.

Difficulty calculators ***CANNOT*** be aware of `ProgressiveCalculationBeatmap` in any way and that is not negotiable.

Reply:
Added those functions to the interface
That's the only other way I see. I initially thought that adding 2 methods only used in diffcalc to `IBeatmap` is worse than adding one check for `ProgressiveCalculationBeatmap` in `OsuDifficultyCalculator`.